ngraph/frontend/onnx_import/src/core/attribute.cpp

```diff
@@ -10,32 +10,22 @@ namespace ngraph
 {
     namespace onnx_import
     {
-        std::vector<Graph> Attribute::get_graph_array(Model& model) const
-        {
-            std::vector<Graph> result;
-            for (const auto& graph : m_attribute_proto->graphs())
-            {
-                result.emplace_back(graph, model);
-            }
-            return result;
-        }
-
         Subgraph Attribute::get_subgraph(const Graph& parent_graph) const
         {
             if (m_attribute_proto->type() != ONNX_NAMESPACE::AttributeProto_AttributeType_GRAPH)
             {
                 throw error::attribute::InvalidData{m_attribute_proto->type()};
             }
 
-            ONNX_NAMESPACE::ModelProto model_proto;
+            auto model_proto = common::make_unique<ONNX_NAMESPACE::ModelProto>();
+
             const auto& graph = m_attribute_proto->g();
-            *(model_proto.mutable_graph()) = graph;
+            *(model_proto->mutable_graph()) = graph;
 
             // set opset version and domain from the parent graph
-            *model_proto.mutable_opset_import() = parent_graph.get_opset_imports();
-
-            Model model{model_proto};
-            return Subgraph{graph, model, parent_graph};
+            *model_proto->mutable_opset_import() = parent_graph.get_opset_imports();
+            auto model = common::make_unique<Model>(model_proto);
+            return Subgraph{model, parent_graph};
```

Comment:
You should call here std::move(parent_graph)
and modify `Graph::Graph` and `Subgraph::Subgraph` to take `std::unique_ptr<Model>` without reference IMO.
I am not sure why it works now, but the ownership of `model` is not transferred to `Subgraph` correctly, and as a result lifetime of `model` is not guaranteed.

Reply:
I decided to replace a l-value reference with a r-value reference. I think that's a most clear solution, is that okay for You?